Split presto into prestodb and prestosql

electrum opened the pull request:
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

There are now two independently maintained [Presto](https://en.wikipedia.org/wiki/Presto_(SQL_query_engine)) projects. Quoting Wikipedia: 

> Development of Presto continues independently with PrestoDB maintained by Facebook and PrestoSQL maintained by the Presto Software Foundation with some cross pollination of code.

To avoid user confusion and to allow usage of either project, this PR renames `presto` to `prestodb` (matching the name it is known by in the community), and creates a new `prestosql` formula.

electrum commented:
It looks like Jenkins fails because the sandboxes already have `prestodb` installed, but these packages cannot both be linked due to using the same binary names.
```
==> Finishing up
The formula built, but is not symlinked into /usr/local
Could not symlink bin/presto
Target /usr/local/bin/presto
is a symlink belonging to prestodb. You can unlink it:
  brew unlink prestodb

To force the link and overwrite all conflicting files:
  brew link --overwrite prestosql

To list all files that would be deleted:
  brew link --overwrite --dry-run prestosql
```

SMillerDev commented:
@Homebrew/core opinions?

MikeMcQuaid commented:
This looks like it makes sense but `presto.rb` likely warrants an upgrade to a newer version at the same time as the rename.

electrum commented:
@MikeMcQuaid Thanks for looking at this. I rebased the PR and `prestodb` is now at the latest version (someone else had already updated it). I also updated `prestosql` to the latest version (there were two releases since I originally created the PR).

electrum commented:
cc @ shigemk2 since you have been maintaining the `presto` formula

electrum commented:
cc @shigemk2 since you have been maintaining the `presto` formula

MikeMcQuaid commented:
> It looks like Jenkins fails because the sandboxes already have `prestodb` installed, but these packages cannot both be linked due to using the same binary names.

This seems undesirable. Is there no situation in which a user would not want to have them both installed? If so they need `conflicts_with`. If not, one could be `keg_only`. Another solution would be renaming one of the scripts or making one `keg_only`.

shigemk2 commented:
I am not sure which we should maintain formula, prestodb or prestosql. I created PR that divides into presto-server and presto-cli in https://github.com/Homebrew/homebrew-core/pull/39151 before, but should we remain prestodb, too?

MikeMcQuaid commented:
Ok so these are separate, conflicting projects (and one succeeds the other?). In that case `conflicts_with` seems appropriate.

MikeMcQuaid commented:
It looks like both projects are still maintained.

electrum commented:
@MikeMcQuaid Thanks for the feedback. I added `conflicts_with` to both formula and confirmed it works correctly when the other is installed:

```
$ brew install --build-from-source prestodb
Error: Cannot install prestodb because conflicting formulae are installed.
  prestosql: because both install `presto` and `presto-server` binaries
```

```
$ brew install --build-from-source prestosql
Error: Cannot install prestosql because conflicting formulae are installed.
  prestodb: because both install `presto` and `presto-server` binaries
```

MikeMcQuaid commented on Formula/prestodb.rb:
```diff
@@ -13,6 +13,8 @@ class Presto < Formula
     sha256 "e7b889a84cbad4dc7fc2f920e37d831cffcbb6bfd67c2f0878d6a5f6cfee02dc"
   end
 
+  conflicts_with "prestosql", :because => "both install `presto` and `presto-server` binaries"
```
Move this as per-`brew style`'s output. `brew style --fix` should do it for you automatically.

electrum replied:
@MikeMcQuaid `brew style` didn't output anything for either formula. However, looking at other formula, it seems to come after `depends_on`, so I moved it there. Please let me know if it should be somewhere else.

electrum commented:
I updated the PR to (hopefully) address @MikeMcQuaid's comment and updated `prestosql` to the latest release.

MikeMcQuaid commented:
Thanks so much for your contribution! Without people like you submitting PRs we couldn't run this project. You rock, @electrum!
